### PR TITLE
add DeepES of PaddleLite

### PR DESCRIPTION
**Pull request by zenghsh3**
- ESAgent of PaddleLite
- CartPole demo

### CODE REVIEW

**Comment by zenghsh3** [comment]
For now, when clone a new ESAgent, will malloc memory of noise and neg_gradients in advance. But neg_gradients are not needed in the sampling stage. Maybe we can design a better way to save memory.

File: deepes/include/torch/es_agent.h [hunk 1]
```diff
@@ -23,36 +24,51 @@
 
 namespace DeepES{
 
-/* DeepES predictor for Torch.
+/* DeepES agent for Torch.
  * Our implemtation is flexible to support any model that subclass torch::nn::Module.
- * That is, we can instantiate a preditor by: predictor = Predcitor<Model>(model);
- * After that, users can clone a predictor for multi-thread processing, add parametric noise for exploration,
+ * That is, we can instantiate a agent by: es_agent = ESAgent<Model>(model);
+ * After that, users can clone a agent for multi-thread processing, add parametric noise for exploration,
```
Comment (TomorrowIsAnOtherDay) [inline deepes/include/torch/es_agent.h:30]: a agent -> `an` agent

File: deepes/include/torch/es_agent.h [hunk 2]
```diff
@@ -23,36 +24,51 @@
 
 namespace DeepES{
 
-/* DeepES predictor for Torch.
+/* DeepES agent for Torch.
  * Our implemtation is flexible to support any model that subclass torch::nn::Module.
- * That is, we can instantiate a preditor by: predictor = Predcitor<Model>(model);
- * After that, users can clone a predictor for multi-thread processing, add parametric noise for exploration,
+ * That is, we can instantiate a agent by: es_agent = ESAgent<Model>(model);
```
Comment (TomorrowIsAnOtherDay) [inline deepes/include/torch/es_agent.h:29]: a agent -> `an` agent

File: deepes/include/torch/es_agent.h [hunk 3]
```diff
@@ -23,36 +24,51 @@
...
-    return new_predictor;
+    std::shared_ptr<ESAgent> new_agent = std::make_shared<ESAgent>();
+    new_agent->set_model(_model, new_model);
+    new_agent->set_sampling_method(_sampling_method);
+    new_agent->set_optimizer(_optimizer);
+    new_agent->set_config(_config);
+    new_agent->set_param_size(_param_size);
+
+    float* new_noise = new float [_param_size];
+    float* new_neg_gradients = new float [_param_size];
```
Comment (TomorrowIsAnOtherDay) [inline deepes/include/torch/es_agent.h:68]: Cloned agents do not need gradients buffers.

File: deepes/include/torch/es_agent.h [hunk 4]
```diff
@@ -80,68 +96,79 @@ class Predictor{
...
+        sampled_tensor_a[j] = tensor_a[j] + _noise[counter + j];
       }
       counter += tensor.size(0);
     }
-    delete[] noise;
     return sampling_key;
   }
 
-  int param_size() {
+  int64_t param_size() {
```
Comment (TomorrowIsAnOtherDay) [inline deepes/include/torch/es_agent.h:171]: This function of paddle version is named as _caculate_param_size. Please update this function name to the same name.

**Pull request closed without merging**